torch/autograd/function.py

```diff
@@ -562,6 +562,11 @@ def bind_default_args(func, *args, **kwargs):
 
         return custom_function_call(cls, *args, **kwargs)
 
+    @staticmethod
+    def _compiled_autograd_key(cls):
+        from torch._dynamo.compiled_autograd import CompiledAutogradCustomFunctionKey
+        return CompiledAutogradCustomFunctionKey.get()
```

Comment:
This is just going to return a constant for all autograd.Functions.  It should be a unique ID *per autograd funciton instance*.

It can just be something close to `id(class_of_autograd_function)` -- but that will miss some corner cases about memory addresses being reused.

Reply:
Updated to use a global autograd function tracker, to assign unique ids as classes are initialized. Is a unique ID per autograd function class sufficient? 

Docs explicitly call out against forward directly, and to use the @classmethod apply, which allocates a new instance every call. If we use a unique ID per instance, there won't be cache hits